### PR TITLE
Feature/group chat

### PR DESCRIPTION
**Pull request by thenderson21**
Fix error in parsing status codes and added null checks for values

### CODE REVIEW

File: Extensions/XEP-0045/MultiUserChat.cs [hunk 1]
```diff
@@ -128,52 +128,57 @@ public bool Input(Im.Message stanza)
...
+			// Service Updates Nick
+			XmlElement xElement = stanza.Data ["x"];
+			if (xElement != null && xElement.NamespaceURI == MucNs.NsUser) {
+				Occupant person = null;
+				foreach (XmlElement item in xElement.GetElementsByTagName ("item")) {
+					// There is only ever one item in a message here but, 
+					// I don't have a better way of getting the first element as an element, not a node.
+					var itemJid = item.GetAttribute ("jid");
+					var itemAffiliation = item.GetAttribute ("affiliation");
+					var itemRole = item.GetAttribute ("role");
```
Comment (jpenny1993) [inline Extensions/XEP-0045/MultiUserChat.cs:152]: 👍

File: Extensions/XEP-0045/MultiUserChat.cs [hunk 2]
```diff
@@ -128,52 +128,57 @@ public bool Input(Im.Message stanza)
...
-                    PrescenceChanged.Raise(this, new GroupPresenceEventArgs(person, statusCodeList));
-                    return true;
-                }
-            }
-
-            // Any message with an Availability status can be managed by the Presence extension
-            return false;
-        }
+        public bool Input (Im.Presence stanza)
+		{
```
Comment (jpenny1993) [inline Extensions/XEP-0045/MultiUserChat.cs:132]: Not sure my upstream will approve of the change in indentation.

**Comment by jpenny1993** [comment]
Cool, I'll complete it when I get out of work :smile:

File: Extensions/XEP-0045/MultiUserChat.cs (hunk 2 above)
Comment (thenderson21) [inline Extensions/XEP-0045/MultiUserChat.cs:132]: Yea, sometime indentations go funky when converting between OSs.

Comment (jpenny1993) [inline Extensions/XEP-0045/MultiUserChat.cs:132]: That's fair :) np

**Pull request merged**